core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -855,12 +857,15 @@ public NativeImageInvokerInfo build() {
                 addExperimentalVMOption(nativeImageArgs, "-H:+AllowFoldMethods");
 
                 /*
+                 * @formatter:off
                  * Foreign Function and Memory API in Native Image, JDK's JEP 454
                  * This is needed for JDK 24+ internal native calls due to AWT,
                  * e.g. JDK-8337237 et al.
-                 *
+                 * Note GraalVM FFI/FFM support per platform:
+                 * https://www.graalvm.org/latest/reference-manual/native-image/native-code-interoperability/foreign-interface/#foreign-functions
+                 * @formatter:on
                  */
-                if (graalVMVersion.compareTo(GraalVM.Version.VERSION_24_2_0) >= 0) {
+                if (graalVMVersion.compareTo(GraalVM.Version.VERSION_24_2_0) >= 0 && AMD64.active) {
```

Comment:
Given that we now handle AArch64 as unsupported, under which configuration do you expect to reach this? I would expect the build to fail sooner than this.

Reply:
I am not sure I follow. It is not AWT specific, it's just that AWT was the first obvious hit. 
AMD64 Linux, Windows and also MacOS are able to use [FFI/FFM](https://www.graalvm.org/latest/reference-manual/native-image/native-code-interoperability/foreign-interface/#foreign-functions).

e.g. if jline hadn't some other issue, this would have made it work on Windows amd64.

Reply:
>  I would expect the build to fail sooner than this.

Build doesn't fail, it just doesn't include the unsupported feature, flag.

Reply:
It's the AWT extension that actively fails the build with an error message *if* you happen to use AWT extension.

Reply:
Uh right we are always enabling FFM not just for AWT. Sorry for the noise.